src/accountsdb/accounts_file.zig

```diff
@@ -55,6 +74,20 @@ pub const AccountInFile = struct {
         rent_epoch: Epoch,
         owner: Pubkey,
         executable: bool,
+
+        pub fn writeToBuf(self: *const AccountInfo, buf: []u8) !usize {
```

Comment:
Since this type doesn't hold any variable length data, using:
```suggestion
        pub fn writeToBuf(self: *const AccountInfo, buf: *[@sizeOf(AccountInfo)]u8) !comptime_int {
```
would allow for fewer failure states statically.

Making the return type `comptime_int` would just require changing the return statement to `return buf.len;`, before which it should be asserted that `offset == buf.len` (if that isn't true something is very wrong).

Reply:
compiler doesnt seem to like that approach - i think it requires AccountInfo to also be comptime known

```
src/core/account.zig:63:35: error: unable to resolve comptime value
        offset += try account_info.writeToBuf(buf[offset..]);
                      ~~~~~~~~~~~~^~~~~~~~~~~
src/core/account.zig:63:35: note: argument to function being called at comptime must be comptime-known
src/accountsdb/accounts_file.zig:78:86: note: expression is evaluated at comptime because the function returns a comptime-only type '@typeInfo(@typeInfo(@TypeOf(accountsdb.accounts_file.AccountInFile.AccountInfo.writeToBuf)).Fn.return_type.?).ErrorUnion.error_set!comptime_int'
        pub fn writeToBuf(self: *const AccountInfo, buf: *[@sizeOf(AccountInfo)]u8) !comptime_int {
                                                                                     ^~~~~~~~~~~~
```

Reply:
Ah right, you need to make the function returning `comptime_int` an `inline fn`, so the result is implicitly comptime-known.

Reply:
In this case I'm not sure it's worth the code bloat just for that, so maybe let's not do that unless we can demonstrate it has worthwhile performance benefits. But it would still be good to make `buf` be a static length buffer.

Reply:
true - tried it out and thought calling it wasnt that great - `fcn(buf[offset...])` is really nice compared to `&buf[offset...offset+54]` - added assert statements that the buffer is big enough

6aac1f1

Reply:
What about just `buf[offset..][0..@sizeOf(AccountInfo)]`?

Reply:
its the same problem of needing to define the sizeOf(AccountInfo) when calling the method unfortunately so would prefer to stick with simple slice